aiohttp/websocket_client.py

```diff
@@ -0,0 +1,166 @@
+    # send request
+    resp = yield from client.request(
+        'get', url, headers=headers, connector=connector, loop=loop)
+
+    # check handshake
+    if resp.status != 101:
+        raise WSServerHandshakeError('Invalid response status')
+
+    if resp.headers.get(hdrs.UPGRADE, '').lower() != 'websocket':
+        raise WSServerHandshakeError('Invalid upgrade header')
+
+    if resp.headers.get(hdrs.CONNECTION, '').lower() != 'upgrade':
+        raise WSServerHandshakeError('Invalid connection header')
+
+    # key calculation
+    key = resp.headers.get(hdrs.SEC_WEBSOCKET_ACCEPT, '')
+    match = base64.b64encode(hashlib.sha1(sec_key + WS_KEY).digest()).decode()
+    if key != match:
+        raise WSServerHandshakeError('Invalid challenge response')
+
+    # websocket protocol
+    protocol = None
+    if protocols and hdrs.SEC_WEBSOCKET_PROTOCOL in resp.headers:
+        resp_protocols = [proto.strip() for proto in
+                          resp.headers[hdrs.SEC_WEBSOCKET_PROTOCOL].split(',')]
+
+        for proto in resp_protocols:
+            if proto in protocols:
+                protocol = proto
+                break
+
+    reader = resp.connection.reader.set_parser(WebSocketParser)
+    writer = WebSocketWriter(resp.connection.writer)
+
+    return ClientWebSocketResponse(reader, writer, protocol, resp, loop)
+
+
+class ClientWebSocketResponse:
+
+    def __init__(self, reader, writer, protocol, response, loop):
+        self._response = response
+        self._conn = response.connection
+
+        self._writer = writer
+        self._reader = reader
+        self._protocol = protocol
+        self._closing = False
+        self._loop = loop
+        self._closing_fut = asyncio.Future(loop=loop)
+
+    @property
+    def closing(self):
+        return self._closing
+
+    @property
+    def protocol(self):
+        return self._protocol
+
+    def ping(self, message='b'):
+        if self._closing:
+            raise RuntimeError('websocket connection is closing')
+        self._writer.ping(message)
+
+    def send_str(self, data):
+        if self._closing:
+            raise RuntimeError('websocket connection is closing')
+        if not isinstance(data, str):
+            raise TypeError('data argument must be str (%r)' % type(data))
+        self._writer.send(data, binary=False)
+
+    def send_bytes(self, data):
+        if self._closing:
+            raise RuntimeError('websocket connection is closing')
+        if not isinstance(data, (bytes, bytearray, memoryview)):
+            raise TypeError('data argument must be byte-ish (%r)' %
+                            type(data))
+        self._writer.send(data, binary=True)
+
+    def close(self, *, code=1000, message=b''):
+        if not self._closing:
+            self._closing = True
+            self._writer.close(code, message)
+
+    @asyncio.coroutine
+    def wait_closed(self):
+        yield from self._closing_fut
+
+    @asyncio.coroutine
+    def receive(self):
```

Comment:
I still think the method name should be the same as for server web socket object.
If you prefer `receive` please add it to `WebSocketResponse` with keeping `receive_msg` as alias (think about backward compatibility).